### PR TITLE
Add more dartdoc to material.dart

### PR DESCRIPTION
**Pull request by abarth**
This patch completes the circuit on this library. The next step is to go
through and add docs to anything we missed.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/material/page.dart [hunk 1]
```diff
@@ -39,9 +39,12 @@ class _MaterialPageTransition extends AnimatedWidget {
   }
 }
 
-const Duration kMaterialPageRouteTransitionDuration = const Duration(milliseconds: 150);
-
+/// A modal route that replaces the entire screen with a material design transition.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/page.dart]: we should say what kind of transition (fading in while sliding up).

(I plan to add another MaterialPageRoute-style class that does a circle wipe transition.)

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 2]
```diff
@@ -29,15 +29,49 @@ const double _kMenuVerticalPadding = 8.0;
...
+/// shows a popup menu, consider using [PopupMenuButton].
+///
+/// See also:
+///
+///  * [PopupMenuItem]
+///  * [PopupMenuDivider]
+///  * [CheckedPopupMenuItem]
+///  * [showMenu]
+///  * [PopupMenuButton]
 abstract class PopupMenuEntry<T> extends StatefulWidget {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/popup_menu.dart]: should probably talk about `T` here

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 3]
```diff
@@ -29,15 +29,49 @@ const double _kMenuVerticalPadding = 8.0;
...
+  /// The amount of vertical space occupied by this entry.
+  ///
+  /// This value must remain constant for a given instance.
   double get height;
+
+  /// The value that should be returned by [showMenu] when the user selects this entry.
   T get value => null;
+
+  /// Whether the user is permitted to select this entry.
   bool get enabled => true;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/popup_menu.dart]: We should probably have PopupMenuDivider override it and set it to false...

Comment (Hixie) [inline packages/flutter/lib/src/material/popup_menu.dart]: maybe the base class shouldn't have a default value, in fact. Just make it abstract.

File: packages/flutter/lib/src/material/popup_menu.dart [hunk 4]
```diff
@@ -52,7 +86,25 @@ class _PopupMenuDividerState extends State<PopupMenuDivider> {
...
+///
+/// To show a popup menu, use the [showMenu] function. To create a button that
+/// shows a popup menu, consider using [PopupMenuButton].
+///
+/// To show a checkmark next to a popup menu item, consider using
+/// [CheckedPopupMenuItem].
+///
+/// See also:
+///
+///  * [PopupMenuItem]
```
Comment (Hixie) [inline packages/flutter/lib/src/material/popup_menu.dart]: this _is_ PopupMenuItem.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**